src/Shared/Components/MaterialHistory/MaterialHistory.component.tsx

```diff
@@ -32,34 +35,35 @@ const MaterialHistory = ({
         }
     }
 
-    const getMaterialHistoryMapWithTime = () => {
-        const historyTimeMap = {}
-
-        material.history?.forEach((history) => {
-            const newDate = history.date.substring(0, 16)
-
-            if (!historyTimeMap[newDate]) {
-                historyTimeMap[newDate] = []
-            }
-            historyTimeMap[newDate].push(history)
-        })
+    const materialHistoryMapWithTime = useMemo(
+        () =>
+            material.history.reduce<Record<string, MaterialHistoryType[]>>((acc, historyElem: MaterialHistoryType) => {
+                const isWebhook = material.type === SourceTypeMap.WEBHOOK
+                const newDate = isWebhook
+                    ? getWebhookDate(material.type, historyElem).substring(0, 16)
+                    : historyElem.date.substring(0, 16)
```

Comment:
Q: What does this substring signify here?

Reply:
Getting date in string,  so extracted out its initials